java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -669,8 +669,11 @@ public final ColumnVector[] slice(int... indices) {
     ColumnVector[] columnVectors = new ColumnVector[nativeHandles.length];
     try {
       for (int i = 0; i < nativeHandles.length; i++) {
-        columnVectors[i] = new ColumnVector(nativeHandles[i]);
+        long nativeHandle = nativeHandles[i];
+        // setting address to zero, so we don't clean it in case of an exception as it
+        // will be cleaned up by the constructor
         nativeHandles[i] = 0;
+        columnVectors[i] = new ColumnVector(nativeHandle);
```

Comment:
So are we guaranteed that when ColumnVector/ColumnView gets the nativeHandle that it will close it on any exception? It looks like there are a few corner cases like OOM in java but also some native calls that could cause exceptions where this is not guaranteed. I'm not sure how much we need to worry about most of them, but it might be nice to fix that while we are here.  Specifically I am thinking that the ColumnView construction should move the try from line 82 up to line 78, right after we set the offHeap state.  The creation of the offHeapState I think we are okay with that failing and we leak a native handle.

Reply:
good point. I have updated